### ISSUE TITLE
Alternate extra wall

### ISSUE THREAD

**Issue by weenan** [bambulab/BambuStudio#4823]
Weakest part of a heated insert is the bond between the infill and the walls of the hole for the insert. Insert pulls out together with the walls separated from the infill.

Orca slicer has a feature called "Alternate extra wall" which prints one extra wall every other layer so the infill is sandwiched in with the wall. This greatly improves the bond between the infill and the wall.

Would be great to have this feature in Bambu Studio.

![291663717-01fdaf1d-ad34-4bc7-a7d0-4871a892d6f5](https://github.com/user-attachments/assets/dbf86f23-14e7-4e15-bf9c-3120be4ecdaf)

**Comment by jensim** [comment]
This feature is amazing! It's enough to have me switch to Orca for prints that require additional strength
One might call this **infill brick layering**, as it gives the print a horizontal brick layering pattern between the infill and wall
It's just such a strong thing, especially when I print with fat nozzles, 1 wall with alternate extra wall is enough

**Comment by Hammerfest** [comment]
Yep, its one of the primary reasons I use OS instead of BBS along with support interface fan control

**Comment by Onecton** [comment]
Cura Slicer has this also dor years now. I really miss that feature in BambuStiudio. I second this!

**Comment by teafusion** [comment]
Also second this _strongly_! I would like to use Orca for that feature, but the organization within which I do my Bambu printing has only Bambu Studio installed and they haven't shown interest in adding Orca installation to their computers.

The feature is already in Orca (and Cura), so what might prevent adding it to the Bambu Studio? I think more advanced users would very much welcome it for functional items printing, those that need the added strength when applicable.

**Comment by S1lentA0** [comment]
Bumping this feature. I noticed that the Alternate Wall feature has disappeared from the latest Bambu Studio version (and the one before that as well IIRC). Do I miss something or reading something wrong, or is this feature actually removed? I can't find anything about it atm.

**Comment by DiCaprio37** [comment]
I'd love this feature as well on linux

**Comment by teafusion** [comment]
> Bumping this feature. I noticed that the Alternate Wall feature has disappeared from the latest Bambu Studio version (and the one before that as well IIRC). Do I miss something or reading something wrong, or is this feature actually removed? I can't find anything about it atm.

Bambu Studio has never had 'Alternate (Extra?) Wall' feature to my reckon, at least not by that name, but from the 'Strength' tab you can still find 'Embedding the wall into the infill' which is Bambu Studio's version of the feature and found right under the 'Wall Loops'. You don't even need to switch on the 'Advanced' mode to be able to use it. Hope this helps!

**Comment by PaulMarsman** [comment]
> Bambu Studio has never had 'Alternate (Extra?) Wall' feature to my reckon, at least not by that name, but from the 'Strength' tab you can still find 'Embedding the wall into the infill' which is Bambu Studio's version of the feature and found right under the 'Wall Loops'. You don't even need to switch on the 'Advanced' mode to be able to use it. Hope this helps!

this is most certainly not doing the same thing as alternate extra wall. Watch https://youtu.be/c7CI6yBTKMc?si=zdEVMeCAi9bFEaMo to see it is applied to all the wall/infill spots. "Embedding the wall" setting is not doing the same thing:

<img width="1901" height="566" alt="Image" src="https://github.com/user-attachments/assets/ff515fc9-e868-405c-806c-57e90d397a62" />

<img width="1901" height="566" alt="Image" src="https://github.com/user-attachments/assets/76c6f5be-c5bc-4884-ac10-32a325ee83d6" />

**Comment by S1lentA0** [comment]
> > Bumping this feature. I noticed that the Alternate Wall feature has disappeared from the latest Bambu Studio version (and the one before that as well IIRC). Do I miss something or reading something wrong, or is this feature actually removed? I can't find anything about it atm.
> 
> Bambu Studio has never had 'Alternate (Extra?) Wall' feature to my reckon, at least not by that name, but from the 'Strength' tab you can still find 'Embedding the wall into the infill' which is Bambu Studio's version of the feature and found right under the 'Wall Loops'. You don't even need to switch on the 'Advanced' mode to be able to use it. Hope this helps!

I just went through some earlier versions of Studio and I couldn't find Alternate Extra Wall. You're right abpout that, I must've been halucinating from the time I was using Orca and Bambu in parallel. But as PaulMarsman already stated, Alternate Extra Wall & Embedding the wall into the Infill are two different kind of parameters, so my request stands.

**Comment by jomixlaf** [comment]
I've been working on porting the feature from Orca to Bambu since I saw Maker's muse video, I'll post here once I'm done 🙃

**Comment by jomixlaf** [comment]
Here you go :) [https://github.com/bambulab/BambuStudio/pull/10370
](url)
We now wait for the Bambu team to may or may not integrate it. 🙏🏻

**Comment by S1lentA0** [comment]
> Here you go :) [https://github.com/bambulab/BambuStudio/pull/10370 ](url) We now wait for the Bambu team to may or may not integrate it. 🙏🏻

Amazing. Such clever people  are out there benefiting the community. Thank you for the effort, let's hope they pick it up and add it.

**Comment by teafusion** [comment]
> Here you go :) [https://github.com/bambulab/BambuStudio/pull/10370 ](url) We now wait for the Bambu team to may or may not integrate it. 🙏🏻

Thank you very much! Would you have any idea why the Bambu Studio team has been deferring the integration of 'Alternate Extra Wall' so long, when it has been in OrcaSlicer for quite awhile already, might it interfere with some other features they have maybe yet to publish or something? I'm glad to have the feature anyway, I used it regularly in Cura back in my UM days, printing mostly just functional objects preferring durability.

**Comment by teafusion** [comment]
> > Bambu Studio has never had 'Alternate (Extra?) Wall' feature to my reckon, at least not by that name, but from the 'Strength' tab you can still find 'Embedding the wall into the infill' which is Bambu Studio's version of the feature and found right under the 'Wall Loops'. You don't even need to switch on the 'Advanced' mode to be able to use it. Hope this helps!
> 
> this is most certainly not doing the same thing as alternate extra wall. Watch https://youtu.be/c7CI6yBTKMc?si=zdEVMeCAi9bFEaMo to see it is applied to all the wall/infill spots. "Embedding the wall" setting is not doing the same thing:

Thank you for your input. Well acknowledged that it's not the same, yet when the 'Embed wall into infill' was introduced in Bambu Studio late last year, it was supposed to be the Bambu Studio substitute for the 'Alternate Extra Wall', without any explanations why that would be so, rather than the actual feature requested here. But lucky we found a guy to port it to Bambu Studio, hope it catches on among the core team eventually.

**Comment by jomixlaf** [comment]
> Thank you very much! Would you have any idea why the Bambu Studio team has been deferring the integration of 'Alternate Extra Wall' so long, when it has been in OrcaSlicer for quite a while already? Might it interfere with some other features they have maybe yet to publish or something? I'm glad to have the feature anyway, I used it regularly in Cura back in my UM days, printing mostly just functional objects, preferring durability.

If you look up there, there are over 5k+ issues open. We can't blame the developer directly because they also have to patch all the bugs, implement company features, test and build their own firmware. They choose what goes where, and I'm okay with that. As a matter of fact, because it is open source, it allows third parties like me to implement features they may not have time for or even on their radar because they have other priorities. I'm glad we can at least improve the software at lightning speed. Imagine if it were closed-source. We wouldn't even have this conversation because they'll probably be 10,000 features behind. 

Some may even do this just as a part-time job, who knows! So far, all my PR as been approved within two weeks. Since this one got a lot of attention and didn't mess up the 1234 files, I have high hopes it will be part of it. 🍻

**Comment by MisterAnderson91** [comment]
Looks like https://github.com/bambulab/BambuStudio/pull/10370 has been merged.

**Comment by teafusion** [comment]
Just upgraded to Bambu Studio 2.7.1.57 and indeed, alternate extra wall was implemented, yay!

Yet the two outer wall loops limitation seems a bit odd. I would understand that 2 wall loops could be a minimum, but that it's a strict limit is beyond me. Why not make it just work with the most inner wall after the two outer wall loops? If there's a technical issue specific for Bambu Studio that prevents using more than one outer wall loops, please someone educate me, I'm curious to learn more.

Nevertheless, if the alternate extra wall will be developed further beyond the two wall restriction, this is a great start and special thanks for you @jomixlaf for being proactive, enabling this to have come into fruition.

**Comment by jomixlaf** [comment]
Hi @teafusion, the word to keep in mind here is recommanded. As it stated when you use more loop 

<img width="171" height="28" alt="Image" src="https://github.com/user-attachments/assets/a4b483bd-0261-4ed6-a6e4-cb4a2dddae91" />

It might work just fine with your model. It might be hit-and-miss. Since I only ported the original feature into Bambu, I can't state why it would be an issue, but I think it's worth investigating, but it would be better in a new thread so people can participate. 

<img width="805" height="649" alt="Image" src="https://github.com/user-attachments/assets/5e03dc57-0c63-4241-8db6-feea87a741f1" />

**Comment by teafusion** [comment]
Thanks for he heads up @jomixlaf ! True, it's only recommendation, but somehow the pop-up with red triangle made it look a 'serious' recommendation or warning even. Would you as more of an expert here be willing to open the new ticket about what you suggested? I think that would be truly worth looking into, like you said.

**Comment by PaulMarsman** [comment]
I believe the warning is just incorrectly set to be that it should be = to 2 walls, not that the minimum should be 2 walls

**Comment by sbrocket** [comment]
FYI for others who might otherwise comment here about it: @teafusion filed #11109 about adjusting the pop-up warning
